src/lib/components/ConfirmTransactions.js

```diff
@@ -33,31 +35,166 @@ export default function ConfirmTransactions(props) {
               <div>
                 <h4>Transaction #{i + 1}</h4>
               </div>
-              <div>
-                <span className="text-secondary">Contract ID: </span>
-                <span className="font-monospace">
-                  {transaction.contractName}
-                </span>
-              </div>
-              <div>
-                <span className="text-secondary">Method name: </span>
-                <span className="font-monospace">{transaction.methodName}</span>
-              </div>
-              {transaction.deposit && transaction.deposit.gt(0) && (
-                <div>
-                  <span className="text-secondary">Deposit: </span>
-                  <span className="font-monospace">
-                    {displayNear(transaction.deposit)}
-                  </span>
-                </div>
-              )}
-              <div>
-                <span className="text-secondary">Gas: </span>
-                <span className="font-monospace">
-                  {displayGas(transaction.gas)}
-                </span>
-              </div>
-              <Markdown text={jsonMarkdown(transaction.args)} />
+
+              {transaction.actions.map((action) => (
+                <>
+                  <div>
+                    <span className="text-secondary">Action type: </span>
+                    <span className="font-monospace">{action.type}</span>
+                  </div>
+
+                  {action.type === "FunctionCall" && (
+                    <>
+                      <div>
+                        <span className="text-secondary">Contract ID: </span>
+                        <span className="font-monospace">
+                          {transaction.receiverId}
+                        </span>
+                      </div>
+                      <div>
+                        <span className="text-secondary">Method name: </span>
+                        <span className="font-monospace">
+                          {action.params.methodName}
+                        </span>
+                      </div>
+                      {action.params.deposit &&
+                        action.params.deposit !== "0" && (
+                          <div>
+                            <span className="text-secondary">Deposit: </span>
+                            <span className="font-monospace">
+                              {displayNear(action.params.deposit)}
+                            </span>
+                          </div>
+                        )}
+                      <div>
+                        <span className="text-secondary">Gas: </span>
+                        <span className="font-monospace">
+                          {displayGas(action.params.gas || TGas.mul(30))}
+                        </span>
+                      </div>
+                      <Markdown text={jsonMarkdown(action.params.args)} />
```

Comment:
Someone asked me to include binary data here. Not sure how we want to handle it. We can always add this later.